framework/src/modules/chain/modules/blocks/process.js

```diff
@@ -537,68 +547,40 @@ Process.prototype.loadBlocksFromPeer = function(peer, cb) {
  * @returns {function} cb - Callback function from params (through setImmediate)
  * @returns {Object} cb.err - Error message if error occurred
  */
-Process.prototype.generateBlock = function(keypair, timestamp, cb) {
+Process.prototype.generateBlock = async function (keypair, timestamp, cb) {
 	// Get transactions that will be included in block
 	const transactions = modules.transactions.getUnconfirmedTransactionList(
 		false,
 		MAX_TRANSACTIONS_PER_BLOCK
 	);
-	const ready = [];
-
-	async.eachSeries(
-		transactions,
-		(transaction, eachSeriesCb) => {
-			modules.accounts.getAccount(
-				{ publicKey: transaction.senderPublicKey },
-				(err, sender) => {
-					if (err || !sender) {
-						return setImmediate(eachSeriesCb, 'Sender not found');
-					}
+	let readyTransactions;
+	try {
+		const transactionIds = await modules.processTransactions.verifyTransactions(
```

Comment:
why do we need to first get transactionIds? Why don't we simply do this: 
```
readyTransactions = await modules.processTransactions.verifyTransactions(
		transactions,				transactions
).filter(response => response.status === TransactionStatus.OK);
```

Reply:
`readyTransactions` needs to be whole transactions, whereas `verifyTransactions` only returns transaction responses. the ids are used in the next line to filter the original transactions using the ids.